tests/tracking/test_tracking.py

```diff
@@ -319,9 +319,9 @@ def test_log_batch():
     for key, value in finished_run.data.metrics.items():
         assert expected_metrics[key] == value
     metric_history0 = client.get_metric_history(run_id, "metric-key0")
-    assert set([(m.value, m.timestamp, m.step) for m in metric_history0]) == set([(1.0, t, 0)])
+    assert set([(m.value, m.timestamp, m.step) for m in metric_history0]) == {(1.0, t, 0)}
```

Comment:
Do we need `set` here?

```suggestion
    assert [(m.value, m.timestamp, m.step) for m in metric_history0] == [(1.0, t, 0)]
```

Reply:
If `metric_history0` contains multiple elements, `set` would make sense.

Reply:
I guess this is for parity with other assertions.

Reply:
an excellent catch. Changed all but the locations where dealing with a `set` makes sense.

Reply:
Had to revert a handful of them due to no ordering guarantees and the simplified runtime of set equivalency compared to recursive list comparisons.